Sources/BuildSystemIntegration/BuiltInBuildSystemAdapter.swift

```diff
@@ -0,0 +1,177 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import BuildServerProtocol
+import Foundation
+import LanguageServerProtocol
+import SKLogging
+import SKOptions
+import SKSupport
+import SwiftExtensions
+import ToolchainRegistry
+
+import struct TSCBasic.AbsolutePath
+import struct TSCBasic.RelativePath
+
+package enum BuildSystemKind {
+  case buildServer(projectRoot: AbsolutePath)
+  case compilationDatabase(projectRoot: AbsolutePath)
+  case swiftPM(projectRoot: AbsolutePath)
+  case testBuildSystem(projectRoot: AbsolutePath)
```

Comment:
I think `testBuildSystem` used to take a `TestBuildSystem` instance instead of a project root but that’s no longer the case. Changed it to a struct named `BuildSystemSpec`.